internal/servers/worker/status.go

```diff
@@ -115,6 +116,35 @@ func (w *Worker) sendWorkerStatus(cancelCtx context.Context) {
 	})
 	if err != nil {
 		w.logger.Error("error making status request to controller", "error", err)
+		// Check for last successful status. Ignore nil last status, this probably
+		// means that we've never connected to a controller, and as such probably
+		// don't have any sessions to worry about anyway.
+		//
+		// If a length of time has passed since we've been able to communicate, we
+		// want to start terminating all sessions as a "break glass" kind of
+		// scenario, as there will be no way we can really tell if these
+		// connections should continue to exist.
+		lastStatus := w.LastStatusSuccess()
+		if lastStatus != nil && lastStatus.StatusTime.Before(time.Now().Add(statusFailGrace*-1)) {
+			w.logger.Warn("status error grace period has expired, canceling all sessions on worker",
+				"last_status_time", lastStatus.StatusTime.String(),
+				"grace_period", statusFailGrace,
+			)
+
+			// Iterate through all sessions. Transition any sessions with state
+			// <SESSIONSTATUS_SESSIONSTATUS_CANCELING to that state.
+			w.sessionInfoMap.Range(func(sessionId, siRaw interface{}) bool {
+				si := siRaw.(*sessionInfo)
+				if si.status < pbs.SESSIONSTATUS_SESSIONSTATUS_CANCELING {
```

Comment:
Please use a switch on the status rather than depending on particular enum values never to change via `<`.

Reply:
:+1: I had a feeling that this might be correct. Will update.